deploy/init-db/db.sql.txt

```diff
@@ -0,0 +1,62 @@
+-- 'resources' is used to store all of our resources. See comments below for an explanation of the columns.
+CREATE TABLE resources (
+    -- example: "/planes/radius/local/resourceGroups/rg1/providers/Applications.Core/applications/my-app"
+    --
+    -- We use columns to break out the most important components of the resource id for optimal querying.
+    --
+    -- Since resource ids are case-insensitive we canonicalize these columns to lowercase.
+    -- We store the original resource id with the original casing so users can work with their preferred
+    -- naming/casing conventions.
+    --
+    -- We ensure a leading and trailing slash on the components of the resource id for ease of comparison.
+    --
+    -- id -> "/planes/radius/local/resourcegroups/rg1/providers/applications.core/applications/my-app/"
+    -- resource_type -> "/applications.core/applications/"
+    -- root_scope -> "/planes/radius/local/resourcegroups/rg1/"
+    -- routing_scope -> "/applications.core/applications/my-app/"
+
+    -- resource id used as key.
+    id TEXT PRIMARY KEY NOT NULL,
+
+    -- original_id is used to store the original id of the resource before any normalization occurs.
+    -- This is provided for compatability with the existing design of the store API, and can be removed
+    -- in the future.
+    original_id TEXT NOT NULL,
+
+    -- resource type by queries to filter by type.
+    resource_type TEXT NOT NULL,
+
+    -- root_scope used by queries to list resources by their scope.
+    root_scope TEXT NOT NULL,
+
+    -- routing_scope used by queries to list resources when they are child resources.
+    routing_scope TEXT NOT NULL,
+    
+    -- etag used for optimistic concurrency control.
+    etag TEXT NOT NULL,
+
+    -- timestamp is used to implement cursor-based pagination (see below).
+    created_at TIMESTAMP (6) WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP,
+
+    -- resource_data stores the resource data.
+    resource_data JSONB NOT NULL
```

Comment:
We can even do JSON queries in future: https://www.postgresql.org/docs/current/functions-json.html#SQLJSON-QUERY-FUNCTIONS.

Like, give me all the resources that belongs to this application/environment.

Reply:
Yes, that's the idea. If you look at the file `postgresclient.go` you'll see a comment to that effect.